### PR TITLE
msi auth: ensure wait for 70 seconds on 410 (IMDS server monthly upgrading)

### PR DESCRIPTION
**Pull request by yugangw-msft**
//CC: @anuchandy

### CODE REVIEW

**Comment by lmazuel** [comment]
@yugangw-msft Your `wait_accumulated` could be inacurate, if the `requests.get` has lags (like connection issues). Let's say for some proxies reason, `requests.get` is taking 10 seconds, `wait_accumulated` does not take this into account, and you will end up waiting 80 seconds.
Would be best if you want absolute 70seconds, to do a `start_time = time.time()` at the beginning, and check `time.time() - start_time > 70`

**Comment by yugangw-msft** [comment]
@lmazuel, I have addressed your feedback. Please take another look and merge it at your convenient. Would be great to do it by Friday morning when @troydai will cut out an CLI installer

File: msrestazure/azure_active_directory.py [hunk 1]
```diff
@@ -757,25 +757,34 @@ def _retrieve_token_from_imds_with_retry(self):
         if self.identity_id:
             payload[self.identity_type] = self.identity_id
 
-        retry, max_retry = 1, 20
+        retry, max_retry, start_time, result = 1, 20, time.time(), None
```
Comment (lmazuel) [inline msrestazure/azure_active_directory.py]: Remove `result`, no pre-declaration needed in Python. This won't pass type validation anyway ([and I'm trying to add that](https://github.com/Azure/msrest-for-python/pull/98)).

**Comment by lmazuel** [comment]
@yugangw-msft Friday is doable. I made a last comment, and I can merge and release. Thanks!

**Comment by yugangw-msft** [comment]
@lmazuel I have addressed your comment. Thanks!

**Pull request merged**